src/swe/physics_swe.c

```diff
@@ -51,14 +51,15 @@ static PetscErrorCode InitMPITypesAndOps(void) {
 
   if (!initialized) {
     // create an MPI data type for the CourantNumberDiagnostics struct
-    const int      num_blocks             = 3;
-    const int      block_lengths[3]       = {1, 1, 1};
-    const MPI_Aint block_displacements[3] = {
+    const int      num_blocks             = 4;
+    const int      block_lengths[4]       = {1, 1, 1, 1};
+    const MPI_Aint block_displacements[4] = {
         offsetof(CourantNumberDiagnostics, max_courant_num),
         offsetof(CourantNumberDiagnostics, global_edge_id),
         offsetof(CourantNumberDiagnostics, global_cell_id),
+        offsetof(CourantNumberDiagnostics, is_set),
     };
-    MPI_Datatype block_types[3] = {MPI_DOUBLE, MPI_INT, MPI_INT};
+    MPI_Datatype block_types[4] = {MPI_DOUBLE, MPI_INT, MPI_INT, MPI_C_BOOL};
```

Comment:
Technically, this field is a `PetscBool`, which is an enum type, and not a C `bool` type, but I think both can be demoted to `int` as needed, so this shouldn't be an issue.

Reply:
`sizeof(bool)` is 1, but `sizeof(PetscBool)` is 4 to match Fortran
```c
#define PetscBool  logical(kind=4)
```
You can use `MPIU_BOOL` to send `PetscBool`.

Reply:
Thanks, Jed!

Reply:
I see `CourantNumberDiagnostics` uses `PetscReal` and `PetscInt`.
```suggestion
    MPI_Datatype block_types[4] = {MPIU_REAL, MPIU_INT, MPIU_INT, MPIU_BOOL};
```